homeassistant/components/konnected.py

```diff
@@ -119,6 +128,33 @@ def device_discovered(service, info):
                             " but not specified in configuration.yaml",
                             discovered.device_id)
 
+    def device_discovered(service, info):
+        """Call when a Konnected device has been discovered."""
+        host = info.get(CONF_HOST)
+        port = info.get(CONF_PORT)
+        setup_device(host, port)
+
+    async def manual_discovery(event):
+        """Init devices on the network with manually assigned addresses."""
+        specified = [dev for dev in cfg.get(CONF_DEVICES) if
+                     dev.get(CONF_HOST) and dev.get(CONF_PORT)]
+
+        while specified:
+            for dev in specified:
+                _LOGGER.debug("Discovering Konnected device %s at %s:%s",
+                              dev.get(CONF_ID),
+                              dev.get(CONF_HOST),
+                              dev.get(CONF_PORT))
+                try:
+                    await hass.async_add_executor_job(setup_device,
```

Comment:
I could do this but I don't see the practical benefit. Yes it would do each call simultaneously, but at the expense of complicating the code a bit. Most users will have only 1 to 3 Konnected devices, and this function is only called once after initial boot of HA, so the practical impact of doing the setup calls one at a time is minimal.

Are you suggesting that this optimization is required at this time to approve this PR for merging?

Reply:
No, I don't think it's required. But we like speed. 🚀 :smile: